docker/docker_image_src_test.go

```diff
@@ -22,3 +23,41 @@ func TestSimplifyContentType(t *testing.T) {
 		assert.Equal(t, c.expected, out, c.input)
 	}
 }
+
+func toDockerRef(t *testing.T, ref string) dockerReference {
+	parsedRef, err := reference.ParseNamed(ref)
+	assert.Equal(t, nil, err)
+	return dockerReference{parsedRef}
+}
+
+func TestRewriteReference(t *testing.T) {
+	ref := toDockerRef(t, "example.com/image:latest")
+
+	out, err := rewriteReference(ref, "example.com", "example.com")
+	assert.Nil(t, err)
+	assert.Equal(t, "example.com/image:latest", out.ref.String())
+}
+
+func TestRewriteReferenceStripPrefix(t *testing.T) {
+	ref := toDockerRef(t, "example.com/foo/image:latest")
+
+	out, err := rewriteReference(ref, "example.com/foo", "example.com")
+	assert.Nil(t, err)
+	assert.Equal(t, "example.com/image:latest", out.ref.String())
+}
+
+func TestRewriteReferenceParseFailed(t *testing.T) {
+	ref := toDockerRef(t, "example.com/image:latest")
+
+	out, err := rewriteReference(ref, "example.com", ":")
+	assert.NotNil(t, err)
+	assert.Nil(t, out)
+}
+
+func TestRewriteReferenceInvalidPrefix(t *testing.T) {
+	ref := toDockerRef(t, "example.com/image:latest")
+
+	out, err := rewriteReference(ref, "foo", "example.com")
+	assert.Nil(t, err)
+	assert.Equal(t, "example.com/image:latest", out.ref.String())
+}
```

Comment:
(This would be easier to extend as a table-driven test, see `TestSimplifyContentType` above, but this works just as well.)

Reply:
I cleaned it up to run within a loop, too.